### DISCUSSION TITLE
Endpoint similar to findAllUserBuyerTransactions in v3?

### DISCUSSION THREAD

**Discussion by kev4ev** [etsy/open-api#959] [Q&A] [answered]
For my use case, I need to get all of a user's transactions and then validate against a particular listing id. I have been looking for something similar to v2's [findAllUserBuyerTransactions](https://www.etsy.com/developers/documentation/reference/transaction#method_findalluserbuyertransactions) but have been unable to find similar functionality in v3. I do see the [getShopReceiptTransactionsByShop](https://developers.etsy.com/documentation/reference#operation/getShopReceiptTransactionsByShop) but it seems this would need to support query params including `user_id` for this to work.

**Am I missing this functionality somewhere or is it a gap between v2 and v3?** If the latter, is it on the roadmap to add it? Thanks!

**Comment by Gareth064** [comment]
What do you mean by a "user's transactions", who is the User in the case? Is it the Shop Owner? The Buyer?

**Reply by Gareth064** [reply, marked as answer]
Also, there is no "public" roadmap that us in the community know about.

Etsy don't actively monitor the discussions in the GitHub anymore, so if you have a question that is best suited for Etsy to answer then you'd need to email them directly. <developer@etsy.com>

**Reply by kev4ev** [reply]
Yes, in my use case the authorized user is the `Buyer`

**Comment by dorkfactory** [comment]
Did you find a solution for this? I'm stuck with the same problem.

Currently I'm able to get the data I need if I'm signed in as the owner of the shop who wants to verify signed in buyers using the "/application/shops/{$shop_id}/transactions" endpoint, so I'm considering just always being signed in as that user somewhere and just periodically updating purchase data for that shop -- but it's a very clunky solution.

**Reply by dorkfactory** [reply]
well, I discovered this Summary of Changes page https://developers.etsy.com/documentation/migration/summary-changes/
findAllUserBuyerTransactions has been "retired" with no explanation of how to get that functionality in v3.

**Comment by emergingdzns** [comment]
I can provide a likely explanation of why this endpoint has been retired. A big effort to secure user data was made on the API last year. The concern was that any developer could begin calling `getUser` and other endpoints like the `findAllUserBuyerTransactions` endpoints and gather potentially private information about any given user. The ability to start a 1 and simply increment up was a problem because bad actors could essentially harvest user identifying information and then all the transactions that they made. Very dangerous and a huge liability for Etsy. I assume that early on in the development of the V3 API, they probably recognized that and needed to close that down.
